lib/Dialect/ESI/capnp/Schema.cpp

```diff
@@ -258,6 +271,161 @@ bool TypeSchemaImpl::operator==(const TypeSchemaImpl &that) const {
   return type == that.type;
 }
 
+//===----------------------------------------------------------------------===//
+// Capnp encode / decode RTL builders.
+//
+// These have the potential to get large and complex as we add more types. The
+// encoding spec is here: https://capnproto.org/encoding.html
+//===----------------------------------------------------------------------===//
+
+static size_t bits(::capnp::schema::Type::Reader type) {
+  using ty = ::capnp::schema::Type;
+  switch (type.which()) {
+  case ty::VOID:
+    return 0;
+  case ty::UINT8:
+  case ty::INT8:
+    return 8;
+  case ty::UINT16:
+  case ty::INT16:
+    return 16;
+  case ty::UINT32:
+  case ty::INT32:
+    return 32;
+  case ty::UINT64:
+  case ty::INT64:
+    return 64;
+  default:
+    assert(false && "Type not yet supported");
+  }
+}
+
+/// Build an RTL/SV dialect capnp encoder for this type.
+Value TypeSchemaImpl::buildEncoder(OpBuilder &b, Value clk, Value valid,
+                                   Value operand) {
+  MLIRContext *ctxt = b.getContext();
+  auto loc = operand.getDefiningOp()->getLoc();
+  ::capnp::schema::Node::Reader rootProto = getTypeSchema().getProto();
+
+  auto i16 =
+      IntegerType::get(ctxt, 16, IntegerType::SignednessSemantics::Signless);
```

Comment:
I think if you leave off the singedness semantics, it defaults to signless. Also, I think there are helpers on the builder to get an IntegerType, like `b.getIntegerType(16)`.